core/src/main/java/bisq/core/btc/wallet/BsqWalletService.java

```diff
@@ -770,12 +770,13 @@ public Transaction getPreparedUnlockTx(TxOutput lockupTxOutput) throws AddressFo
     // Addresses
     ///////////////////////////////////////////////////////////////////////////////////////////
 
-    private LegacyAddress getChangeAddress() {
+    private Address getChangeAddress() {
         return getUnusedAddress();
     }
 
-    public LegacyAddress getUnusedAddress() {
-        return (LegacyAddress) wallet.getIssuedReceiveAddresses().stream()
+    public Address getUnusedAddress() {
+        return wallet.getIssuedReceiveAddresses().stream()
+                .filter(address -> Script.ScriptType.P2WPKH.equals(address.getOutputScriptType()))
```

Comment:
So we enforce a new segwit address in case a unused legacy address would be available. Just want to double check that this was your intention. I agree that this makes sense so no old dangling legacy addresses will be used.

Reply:
Yes, that was my intention. The default active keychain (that is, the segwit one) comes last in the list of keychains queried by bitcoinj for issued addresses, so any unused legacy addresses would show up first in the filtered list. That was also what I observed on regtest, if I recall correctly.